### ISSUE TITLE
What do these options do?

### ISSUE THREAD

**Issue by mwoehlke-kitware** [uncrustify/uncrustify#1880]
As you may know from #1852, I am trying to revise the documentation of our options. However, I'm having trouble with a few:

- `nl_(before|after)_func_*`: When exactly do these apply? What is a "function class (prototype|definition)"?
- `cmt_indent_multi`: What exactly does this do or not do? What are "keyword substitution" and "leading chars" in this context?

Any help with writing a clear and consistent definition for these would be appreciated!

**Comment by CDanU** [comment]
proto = decl

----------
```cpp
void f();
void g();
```
```sql
nl_after_func_proto = 2
```
```cpp
void f();

void g();
```

----------

```cpp
void f();
void g();
int i;
```
```sql
nl_after_func_proto_group = 2
```
```cpp
void f();
void g();

int i;
```
----------

> CT_FUNC_CLASS_PROTO, // ctor or dtor for a class

```cpp
class c
{
public:
c();
~c(  );
c(int i);
}
```
```sql
nl_after_func_class_proto = 2 #such an oddball option 
```
```cpp
class c
{
public:
c();

~c(  );

c(int i);

}
```

------


.....

**Comment by mwoehlke-kitware** [comment]
>> nl_(before|after)_func_*: When exactly do these apply? What is a "function class (prototype|definition)"?
>
> proto = decl ... unless its not and proto = ctor/dtor

Okay, after poking around in the guts of this stuff with several of my recent PR's, I think I've got it sorted. These deal with `CT_FUNC_{PROTO|DEF}` vs. `CT_FUNC_CLASS_{PROTO|DEF}`; the latter, as noted, refer to ctors/dtors as opposed to regular methods (or operators, or free functions). I've made updates to my local branch for #1852, which I think just leaves `cmt_indent_multi`. (I believe I understand now what that does, but don't yet know how to rephrase it...)
